core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -153,6 +153,49 @@ func (provider *VersionedDBProvider) Close() {
 	provider.redoLoggerProvider.close()
 }
 
+// Drop drops the couch dbs and redologger data for the channel.
+// It is not an error if a database does not exist.
+func (provider *VersionedDBProvider) Drop(channelName string) error {
+	metadataDBName := constructMetadataDBName(channelName)
+	couchDBDatabase := couchDatabase{couchInstance: provider.couchInstance, dbName: metadataDBName, indexWarmCounter: 1}
+	_, couchDBReturn, err := couchDBDatabase.getDatabaseInfo()
+	if couchDBReturn != nil && couchDBReturn.StatusCode == 404 {
```

Comment:
we should avoid propagating the status code at the level of statecouchdb. We can change the function `couchDBDatabase.getDatabaseInfo()` signature to return `(dbInfo, error)` and return `nil, nil` for  `StatusCode == 404`. But since, it may affect other places where `getDatabaseInfo()` is getting used, you can fix this in a separate PR

Reply:
Will be done in FAB-18061